cmd/collector/app/collector.go

```diff
@@ -169,7 +177,18 @@ func (c *Collector) Close() error {
 		defer cancel()
 	}
 
-	// by now, we shouldn't have any in-flight requests anymore
+	// by now, we shouldn't have any in-flight requests anymore, close the processor and handlers
+	for _, closer := range []io.Closer{c.spanProcessor, c.zipkinSpansHandler, c.jaegerBatchesHandler, c.grpcHandler} {
+		if closer == nil {
+			continue
+		}
+
+		if err := closer.Close(); err != nil {
+			c.logger.Error("Failed to close span handler.", zap.Error(err))
```

Comment:
* start with lowercase
* first item in the list is technically not a span handler